doc/content/xapi-guard/_index.md

```diff
@@ -26,3 +28,39 @@ Xapi-guard forwards calls from domains to xapi to persist UEFI variables, and up
 To do this, it listens to 1 socket per service (varstored, or swtpm) per domain.
 To create these sockets before the domains are running, it listens to a message-switch socket.
 This socket listens to calls from xenopsd, which orchestrates the domain creation.
+
+To protect the domains from xapi being unavailable transiently, xapi-guard provides an on-disk cache for vTPM writes.
+This cache stores the requests temporarily until xapi can be contacted again.
+This situation usually happens when xapi is being restarted as part of an update.
+
+The cache module consists of two Lwt threads, one that writes to disk, and another one that reads from disk.
+Both try to send the requests to xapi, depending on the state, to avoid using the cache if possible.
+The threads communicate through a bounded queue, this is done to limit the amount of memory used.
+
+This means that the cache operates in three modes:
+- Direct: during normal operation the disk is not used at all
+- Engaged: both threads use the queue to order events
+- Disengaged: A thread dumps request to disk while the other reads the cache
+  until it's empty
+
+```mermaid
+---
+title: Cache State
```

Comment:
style: suggesting a digram with states inside the box, and the explanation of the state somewhere else so that people can first understand the high level picture of the state machine, and then dive into what each state means. Something like this:
```
---
title: Cache State
---
stateDiagram-v2

    Disengaged
    note right of Disengaged
         Writer doesn't add requests to queue. 
         Reader reads from cache and tries to push to xapi
    end note
    Direct
    note left of Direct 
        Writer bypasses cache, send to xapi. Reader waits.
    end note
    Engaged 
    note right of Engaged
        Writer writes to cache and adds requests to queue. 
        Reader reads from queue and tries to push to xapi.
    end note
    [*] --> Disengaged
    Disengaged --> Disengaged : Reader pushed pending TPMs to xapi, in the meantime TPMs appeared in the cache
    Disengaged --> Direct : Reader pushed pending TPMs to xapi, cache is empty
    Direct --> Direct : Writer receives TPM, sent to xapi
    Direct --> Engaged : Writer receives TPM, error when sent to xapi
    Engaged --> Direct : Reader sent TPM to xapi, finds an empty queue
    Engaged --> Engaged : Writer receives TPM, queue is not full
    Engaged --> Disengaged : Writer receives TPM, queue is full
    ```
    
![image](https://github.com/xapi-project/xen-api/assets/17672534/23f37352-707e-48b4-abfb-1c17eed22053)

Reply:
although IMO the diagram above is not very aesthetic either so up to you

Reply:
better than the graph that's being displayed now, which is illegible...